### PR TITLE
More passing console tests

### PR DESCRIPTION
**Pull request by pfgithub**
### What does this PR do?

- ✓ test-console-group
- ✓ test-console-instance
- ✓ test-console-no-swallow-stack-overflow
- ✓ test-console-tty-colors
- ✓ test-console-with-frozen-intrinsics
- ✗ test-console-clear: overwrites process.stdout
- ✗ test-console-count: overwrites process.stdout
- ✗ test-console-formatTime: tests time printing which is from internal/util/debuglog
- ✗ test-console-stdio-setters: overwrites process._stdout
- ✗ test-console-table: bun centers fields, but node left-aligns fields
- ✗ test-console: requires implementation of `console._times` & overwrites process.stdout with hijackstdio

### CODE REVIEW

**Comment by robobun** [comment]
@pfgithub, your commit d7d0f6aab5d824436c7de219715fed4b00699abc is building: [`#7682`](https://buildkite.com/bun/bun/builds/7682)

File: src/js/builtins/ConsoleObject.ts [hunk 1]
```diff
@@ -289,28 +292,35 @@ export function createConsoleConstructor(console: typeof globalThis.console) {
     } = options;
 
     if (!stdout || typeof stdout.write !== "function") {
-      // throw new ERR_CONSOLE_WRITABLE_STREAM("stdout");
-      throw new TypeError("stdout is not a writable stream");
+      const err = new TypeError("stdout is not a writable stream");
+      err.code = "ERR_CONSOLE_WRITABLE_STREAM";
```
Comment (Jarred-Sumner) [inline src/js/builtins/ConsoleObject.ts]: Please add these to ErrorCode.ts

```suggestion
      throw $ERR_CONSOLE_WRITABLE_STREAM("stdout is not a writable stream");
```

File: src/js/builtins/ConsoleObject.ts [hunk 2]
```diff
@@ -289,28 +292,35 @@ export function createConsoleConstructor(console: typeof globalThis.console) {
...
+      validateInteger(groupIndentation, "groupIndentation", 0, kMaxGroupIndentation);
     }
 
     if (inspectOptions !== undefined) {
-      // validateObject(inspectOptions, "options.inspectOptions");
+      validateObject(inspectOptions, "options.inspectOptions");
 
       if (inspectOptions.colors !== undefined && options.colorMode !== undefined) {
-        // throw new ERR_INCOMPATIBLE_OPTION_PAIR("options.inspectOptions.color", "colorMode");
+        const err = new TypeError(
```
Comment (Jarred-Sumner) [inline src/js/builtins/ConsoleObject.ts]:
```suggestion
        const err = $ERR_INCOMPATIBLE_OPTION_PAIR(
```

**Comment by Jarred-Sumner** [comment]
test-console-tty-colors is probably a CI env var conflict

**Pull request merged**